tests/Test.ADAL.NET.Integration/SovereignCloudsTests.cs

```diff
@@ -61,15 +97,103 @@ public void Initialize()
+
+            // Assure instance cache is empty
+            InstanceDiscovery.InstanceCache.Clear();
+            Assert.AreEqual(0, InstanceDiscovery.InstanceCache.Count());
+
+            // Creating AuthenticationContext with common Authority
+            var authenticationContext =
+                new AuthenticationContext(TestConstants.DefaultAuthorityCommonTenant, false, new TokenCache());
+
+            // Mock value for authentication returnedUriInput, with cloud_instance_name claim
+            var authReturnedUriInputMock = TestConstants.DefaultRedirectUri + "?code=some-code" + "&" +
+                                           TokenResponseClaim.CloudInstanceHost + "=" + sovereignAuthorityHost;
+
+            MockHelpers.ConfigureMockWebUI(
+                new AuthorizationResult(AuthorizationStatus.Success, authReturnedUriInputMock),
+                // validate that authorizationUri passed to WebUi contains instance_aware query parameter
+                new Dictionary<string, string> { { "instance_aware", "true" } });
+
+            // German authority not included
+            HttpMessageHandlerFactory.AddMockHandler(MockHelpers.CreateInstanceDiscoveryMockHandler(TestConstants.GetDiscoveryEndpoint(TestConstants.DefaultAuthorityCommonTenant), content));
+            
+            HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler
+            {
+                Method = HttpMethod.Post,
+                ResponseMessage =
+                    MockHelpers.CreateSuccessTokenResponseMessage(TestConstants.DefaultUniqueId,
+                    TestConstants.DefaultDisplayableId, TestConstants.DefaultResource),
+
+                AdditionalRequestValidation = request =>
+                {
+                    // make sure that Sovereign authority was used for Authorization request
+                    Assert.AreEqual(sovereignAuthorityHost, request.RequestUri.Authority);
+                }
+            });
+
+            // German authority not included
+            HttpMessageHandlerFactory.AddMockHandler(MockHelpers.CreateInstanceDiscoveryMockHandler(TestConstants.GetDiscoveryEndpoint(TestConstants.DefaultAuthorityBlackforestTenant), content));
+
+            var authenticationResult = 
+                await authenticationContext.AcquireTokenAsync(TestConstants.DefaultResource,
+                TestConstants.DefaultClientId,
+                TestConstants.DefaultRedirectUri, platformParameters, UserIdentifier.AnyUser, "instance_aware=true");
+
+            Assert.AreEqual(5, InstanceDiscovery.InstanceCache.Count());
+
+            // One mock remaining
+            Assert.AreEqual(1, HttpMessageHandlerFactory.MockHandlersCount());
+
+            // German authority not included in instance cache
+            Assert.AreEqual(false, InstanceDiscovery.InstanceCache.Keys.Contains("login.microsoftonline.de"));
```

Comment:
why is this false? DE instance should have been added to the cache

Reply:
it's false because the instance discover call hasn't happened yet. Just verifying that the DE instance is not in the instance cache.